### ISSUE TITLE
No 'zoneinfo' AuthUserAttributeKey 

### ISSUE THREAD

**Issue by najeeburrehman-brainx** [aws-amplify/amplify-ios#751]
I need to send the timezone of the user in the signup attributes because my current backend implementation require user timezone against the key **zoneinfo** that is available on android but I can't find this key in AuthUserAttributeKey enum.

I tried to send it as custom attribute but it fails with error **"A client attempted to write unauthorized attribute"**.

1. Is it intended to not have the key for zoneinfo in AuthUserAttributeKey as it exists in android ?

2. What should be the workaround to send the timezone in userattribute during signup ?

Any help would be highly appreciated.
Thanks.

**Comment by lawmicha** [comment]
Hi @najeeburrehman-brainx, thanks fo reporting this issue. On Android, there's [zone info attribute key](https://github.com/aws-amplify/amplify-android/blob/12ec8e9819228f63bd677a0431a5ab191f79d9e1/core/src/main/java/com/amplifyframework/auth/AuthUserAttributeKey.java#L40) as mentioned. 

on iOS, it looks like it is missing a couple compared to the [docs](https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-settings-attributes.html). ([Related code](https://github.com/aws-amplify/amplify-ios/blob/eb5b7ff98ae9ac090dc8c71e374b8f3bcb0ee472/Amplify/Categories/Auth/Models/AuthUserAttribute.swift#L34))
- zoneinfo
- website 
- updated_at
- profile

There is an intention to not have certain keys defined at the Amplify layer to avoid any Cognito specific attributes, and we'll have to check in with Android to understand which ones should and should not be there for parity.

As for the workaround (or permanent solution) you can use the `unknown("zoneinfo")` as the attribute key.

**Comment by najeeburrehman-brainx** [comment]
Thanks for your response.
I have tried the zoneinfo with custom attribute key i.e. `.custom("zoneinfo")` but it didn't work and even gives the error to write the unauthorized attribute.
Let me give a chance to `.unknown("zoneinfo")` as you suggest then I'll update you.
Thanks again.

**Comment by lawmicha** [comment]
`.custom("zoneinfo")` will translate to "custom:zoneinfo", i believe for actual "custom" user pool attributes that you add to the user pool, ([docs](https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-settings-attributes.html#user-pool-settings-custom-attributes]))

**Comment by najeeburrehman-brainx** [comment]
Yupp it works with `.unknown("zoneinfo")`.
Thank you.

**Comment by traysonkelii** [comment]
> `.custom("zoneinfo")` will translate to "custom:zoneinfo", i believe for actual "custom" user pool attributes that you add to the user pool, ([docs](https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-settings-attributes.html#user-pool-settings-custom-attributes%5D))

(Android specific)
I would just like to clarify my findings regarding this statement. There are two parts to enabling custom attributes created in user pools. 

First: You must enable read/write privileges on custom created roles via AWS Console (General settings -> App clients -> Show details -> Set attribute read and write permissions link). 

Second: `AuthUserAttributeKey.custom()` is the function used for accessing/sending values to these custom fields, however, the string `custom:` MUST be prepended to your custom field. My assumption was that this would be excluded given the function call name. A little misleading but I hope this helps someone else out there. 

TLDR;

This threw an error:
```
AuthUserAttributeKey.custom("example");
```
This worked as expected:
```
AuthUserAttributeKey.custom("custom:example");
```
